### DISCUSSION TITLE
RequestedHeartbeat not ablet to set from .Net

### DISCUSSION THREAD

**Discussion by manojnctks** [rabbitmq/rabbitmq-dotnet-client#1651] [Q&A]
I am using **RabbitMq ver 3.13.6** with _**.net core7.0 (C#).**_ I am trying to set the hearbet timeout value which is by default 60se. To set the value I have tried two ways:
1) Setting the property RequestedHeartbeat value at the time of creating the factory object using ConnectionFactory class.
var cf = new ConnectionFactory{RequestedHeartbeat =TimeSpan.FromMinutes(10);......}
2) I created the factory object using ConnectionFactory

**var cf=new ConnectionFactory();
cf.RequestedHeartbeat  = TimeSpan.FromMinutes(10);**

In both scenario later on when I create 
**IConnection connection = cf.CreteConnection()**
and try to check the of 
**connection.Heartbeat**
I always get the default value i.e 60 sec, even though I explicitly set some other value.

I followed the .net guideline given in url [https://www.rabbitmq.com/docs/heartbeats#heartbeats-timeout](url) but not work for me, or I am missing something.

Kindly suggest how to set the value of RequestedHeartbeat as per my requirement
**Note: My project is using .Net Core 7**
Thanks in adavance

**Comment by lukebakken** [comment]
https://github.com/rabbitmq/rabbitmq-server/blob/main/COMMUNITY_SUPPORT.md#who-is-eligible-for-community-support

Please provide code that I can clone, compile and run to see _EXACTLY_ what you are trying. I can't provide assistance otherwise. Thanks.

In addition, please state _why_ you wish to change the heartbeat timeout value.

**Comment by michaelklishin** [comment]
Note that a heartbeat timeout of 10 minutes is as good as not having any. In which case you might as well disable them and use [TCP keepalives](https://www.rabbitmq.com/docs/heartbeats#tcp-keepalives) if that's an option.

Values between 15s and 60s are highly recommended.
